classes/migration/upgrade/v3_4_0/I7167_RemoveDuplicatedUserSettingsAndDeprecatedFields.php

```diff
@@ -90,6 +91,9 @@ public function up(): void
                     AND best.user_id IS NOT NULL"
             );
         }
+
+        // Drop the deprecated fields
+        Schema::table('user_settings', fn (Blueprint $table) => $table->dropColumn('assoc_id', 'assoc_type'));
```

Comment:
This causes a fatal error in the SciELO database, which does not have an `assoc_id` column in `user_settings`.

Reply:
Correction: The SciELO database has `assoc_id` and `assoc_type` columns; the problem is that they are both included in the `user_settings_pkey` index, and can't be dropped before the index is cleaned up. (This should be common to any database up to and including 3.3.0, I think -- I'm not sure how the upgrade tests are passing; maybe some versions of MySQL/MariaDB are OK with dropping columns that are used in indexes, and mine isn't?)

Reply:
This index cleanup was here before, you've moved it to another place, but forgot to bring the removal of fields, so this is ok to include at a later time.